### PR TITLE
cmds/core/chmod: fix issue with multiple files

### PR DESCRIPTION
**Pull request by binjip978**
if any file is not exists or some other error chmod will return and not process other files in args. Add stderr to print all the errors and remove mode as return value.

### CODE REVIEW

File: cmds/core/chmod/chmod.go [hunk 1]
```diff
@@ -135,54 +148,61 @@ func calculateMode(modeString string) (mode os.FileMode, octval uint64, mask uin
...
 			})
 			if err != nil {
-				return mode, err
+				finalErr = err
+				fmt.Fprintln(c.stderr, err)
 			}
 		} else {
-			if mode, err = changeMode(name, mode, octval, mask); err != nil {
-				return mode, err
+			err := changeMode(name, mode, octval, mask)
```
Comment (rminnich) [inline cmds/core/chmod/chmod.go:198]: would it be worth using errors.Join here, so we get them all? That might require moving Go forward but ... ?

File: cmds/core/chmod/chmod.go [hunk 2]
```diff
@@ -191,7 +211,7 @@ func main() {
 		reference = flag.String("reference", "", "use mode from reference file")
 	)
 	flag.Parse()
-	if _, err := chmod(*recursive, *reference, flag.Args()...); err != nil {
-		log.Fatal(err)
+	if err := command(os.Stderr, *recursive, *reference).run(flag.Args()...); err != nil {
+		os.Exit(1)
```
Comment (rminnich) [inline cmds/core/chmod/chmod.go:215]: if you did use errors.Join, above, this change might not be needed?

**Review by rminnich — COMMENTED** [review]
very nice. Just had a question on errors.

I am lost as to why that function returned the mode.

File: cmds/core/chmod/chmod.go (hunk 2 above)
Comment (binjip978) [reply cmds/core/chmod/chmod.go:215]: Even with error.Join if we want to the same behaviour as regular chmod we need to change log.Fatal to something like fmt.Sprintf(os.Stderr, errors) and that call os.Exit(1)

**Comment by binjip978** [comment]
> very nice. Just had a question on errors.
> 
> I am lost as to why that function returned the mode.

I actually don't know, but maybe it was handy for testing.

File: cmds/core/chmod/chmod.go (hunk 2 above)
Comment (rminnich) [reply cmds/core/chmod/chmod.go:215]: somehow my comment on this got lost. 
https://go.dev/play/p/UMgj4neXlI6
so you can, given something constructed with errors.Join
log.Fatal(err)
and that works.

Comment (binjip978) [reply cmds/core/chmod/chmod.go:215]: ok, lets do it with errors.Join, looks like CI is already moved beyond 1.19

Comment (binjip978) [reply cmds/core/chmod/chmod.go:215]: I think no error.Join until February :)

# github.com/u-root/u-root/cmds/core/chmod
core/chmod/chmod.go:194:19: undefined: errors.Join
core/chmod/chmod.go:199:19: undefined: errors.Join
note: module requires Go 1.21

Exited with code exit status 2

So probably it make sense to merge this as is and cleanup a bit after we abandon 1.19

Comment (rminnich) [reply cmds/core/chmod/chmod.go:215]: arg, so annoying. you are right.

**Review by rminnich — APPROVED** [review]

**Pull request merged**